kf_lib_data_ingest/etl/transform/guided.py

```diff
@@ -45,44 +59,28 @@ def _apply_transform_funct(self, data_dict):
 
         # Apply user supplied transform function
         transform_funct = self.transform_module.transform_function
-        entity_df_dict = transform_funct(df_dict)
-
-        # -- Validation of transform function output --
-        # Must return a dict of dataframes
-        assert_safe_type(entity_df_dict, dict)
-        assert_all_safe_type(entity_df_dict.values(), pandas.DataFrame)
-
-        # Keys must be valid target service entities
-        target_concepts = set(self.target_api_config.concept_schemas.keys())
-        error = KeyError(f"Error in transform_function's "
-                         f"return value, file: {filepath}. Keys in dict "
-                         "must be valid target_concepts. Must be one of: \n"
-                         f"{pformat(target_concepts)}")
-        if not entity_df_dict.keys():
-            raise error
-        for key in entity_df_dict.keys():
-            if key not in target_concepts:
-                raise error
-
-        return entity_df_dict
-
-    def _standard_to_target(self, entity_df_dict):
+        merged_df = transform_funct(df_dict)
+
+        # Validation of transform function output
+        assert_safe_type(merged_df, pandas.DataFrame)
+        self.transform_func_df = merged_df
```

Comment:
If you move this assignment of `self.transform_func_df` down to replace the left side of the `all_data_df = _apply_transform_funct` line in `_do_transform`, then I think the motivation for storage becomes a bit clearer.

Reply:
👍